README.md

```diff
@@ -0,0 +1,160 @@
+might not error and correctness is at stake.
+
+### Minimize administrative helpers
+
+Administrative helpers are methods for admin tasks, like user creation.
+These are rarely used and have maintenance costs as the server changes
+the administrative API. Don't create administrative helpers; let users
+rely on "RunCommand" for administrative commands.
+
+### Check wire version, not server version
+
+When determining server capabilities within the driver, rely only on the
+maxWireVersion in the hello response, not on the X.Y.Z server version.
+An exception is testing server development releases, as the server bumps
+wire version early and then continues to add features until the GA.
+
+### When in doubt, use "MUST" not "SHOULD" in specs
+
+Specs guide our work. While there are occasionally valid technical
+reasons for drivers to differ in their behavior, avoid encouraging it
+with a wishy-washy "SHOULD" instead of a more assertive "MUST".
+
+### Defy augury
+
+While we have some idea of what the server will do in the future, don't
+design features with those expectations in mind. Design and implement
+based on what is expected in the next release.
+
+Case Study: In designing OP_MSG, we held off on designing support for
+Document Sequences in Replies in drivers until the server would support
+it. We subsequently decided not to implement that feature in the server.
+
+### The best way to see what the server does is to test it
+
+For any unusual case, relying on documentation or anecdote to anticipate
+the server's behavior in different versions/topologies/etc. is
+error-prone. The best way to check the server's behavior is to use a
+driver or the shell and test it directly.
+
+### Drivers follow semantic versioning
+
+Drivers should follow X.Y.Z versioning, where breaking API changes
+require a bump to X. See [semver.org](https://semver.org/) for more.
+
+### Backward breaking behavior changes and semver
+
+Backward breaking behavior changes can be more dangerous and disruptive
+than backward breaking API changes. When thinking about the implications
+of a behavior change, ask yourself what could happen if a user upgraded
+your library without carefully reading the changelog and/or adequately
+testing the change.
+
+## Writing Documents
+
+Write documents using
+[reStructuredText](http://docutils.sourceforge.net/rst.html), following
+the [MongoDB Documentation Style
+Guidelines](https://www.mongodb.com/docs/meta/style-guide/).
+
+Store all source documents in the `source/` directory.
+
+## Linting
+
+This repo uses [pre-commit](https://pypi.org/project/pre-commit/) for
+managing linting. `pre-commit` performs various checks on the files and
+uses tools that help follow a consistent style within the repo.
+
+To set up `pre-commit` locally, run:
+
+```bash
+brew install pre-commit
+pre-commit install
+```
+
+To run `pre-commit` manually, run `pre-commit run --all-files`.
+
+To run a manual hook like `rstcheck` manually, run:
```

Comment:
Is this still relevant? I think it'd be forgotten if left until the last spec is migrated to Markdown.

Reply:
Updated